src/TBLIS.jl

```diff
@@ -1,48 +1,143 @@
 module TBLIS
 
-using Libdl
-using LinearAlgebra
-using Hwloc_jll
 using tblis_jll
 
-using LinearAlgebra: BlasFloat
-import LinearAlgebra: mul!
-
-export TTensor
-export mul!
-export add!
-
-global tblis = C_NULL
-global tci = C_NULL
-global hwloc = C_NULL
-
-function __init__()
-    if Sys.isapple()
-        #global tblis = dlopen(joinpath(dirname(pathof(TBLIS)),"libtblis"))
-        global tblis = dlopen(joinpath(@__DIR__,"libtblis"))
-    else
-        global tci = dlopen(tblis_jll.tci)
-        global hwloc = dlopen(Hwloc_jll.libhwloc)
-        global tblis = dlopen(tblis_jll.tblis)
+# Exports
+export tblis_tensor, tblis_scalar
+export tblis_tensor_add, tblis_tensor_mult, tblis_tensor_dot
+export tblis_get_num_threads, tblis_set_num_threads
```

Comment:
What do you think about removing the `tblis_` prefix, not exporting these, and have a public interface `TBLIS.tensor`, `TBLIS.scalar`, etc.?

Reply:
I think this is also how CUDA or cuTENSOR does it if I am not mistaken, so I would be completely fine with this. My only objection would be that I really prefer to avoid having to maintain a lot (any) functionality here, and changing the function names means that they no longer coincide with the documentation pages of TBLIS (which admittedly are quite terse). In any case, users would ideally never have to interface with any of the functionality here, and rather work with overarching packages, so the longer function names seem not too bothersome. Ultimately, I don't either option makes a big difference, so I'm okay with anything

Reply:
As mentioned in discussion with @lkdvos over at TensorOperationsTBLIS, I would love to be able to write

```julia
TBLIS.get_num_threads()
TBLIS.set_num_threads(n)
```
in analogy with
```julia
BLAS.get_num_threads()
BLAS.set_num_threads(n)
Strided.get_num_threads()
Strided.set_num_threads(n) # not completely fair as this one does now spawn its own threads but can only use Julia threads, but anyway
```

That is not really an argument though, since `tblis_get/set_num_threads` uses unsigned integers, and therefore has an oddly looking output. So we probably want to add one layer above the wrapper methods that does the conversion to standard `Int`, and we can then use `TBLIS.g/set_num_threads` for that.

Reply:
Ok, in that case, for aesthetic reasons, let me just swap everything out for `TBLIS.f` (without exporting anything).

Reply:
Any thoughts on the other functions? I.e. `tblis_tensor_add` vs. `TBLIS.tensor_add`?

Your comment makes me think it is better to start out sticking with `tblis_tensor_add`, etc. in case there are quirks with the TBLIS interface that we want to fix up with a higher layer of functions, and we could reserve the nicer syntax `TBLIS.tensor_add` (or probably `TBLIS.tensoradd!` to match `TensorOperations.tensoradd!`) for that higher layer. So the `tblis_` functions would be lower level functions that are closer to the TBLIS C interface (like what @lkdvos was suggesting above).

Reply:
Wait, but then the `TBLIS.get_num_threads` becomes the current `tblis_get_num_threads` that returns a `Cuint` rather than a layer above that converts to regular Int?

Reply:
Okay, so I missed these last couple comments

Reply:
To clarify, I am back to being convinced that the functions @lkdvos wrote originally should still be called `tblis_f` to indicate that they are following the TBLIS C API, and then we can add on layers above those called `TBLIS.f` as needed, for example `TBLIS.get_num_threads` written in terms of `tblis_get_num_threads` as @Jutho was suggesting.

Reply:
So does everyone agree with:
- I keep mirroring the TBLIS C API, having `tblis_f` as function names
- I add `TBLIS.set_num_threads` and `TBLIS.get_num_threads` to fix the weird Int problem